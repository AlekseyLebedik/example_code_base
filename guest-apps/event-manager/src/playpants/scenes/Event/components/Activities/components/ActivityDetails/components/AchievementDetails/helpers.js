import startCase from 'lodash/startCase';
import isInteger from 'lodash/isInteger';
import capitalize from 'lodash/capitalize';

const timeFields = [
  'Last Update Timestamp',
  'Activation Timestamp',
  'Creation Timestamp',
  'Code Signature Timestamp',
];

const boolFields = ['Is Active'];

export const startCaseKeys = target => {
  if (!target) return target;
  const startCased = {};
  Object.entries(target).forEach(([key, value]) => {
    startCased[startCase(key)] = value;
  });
  return startCased;
};

export const formatRuleset = (ruleset, formatDateTime) => {
  if (!ruleset) return ruleset;
  const { code, ...rest } = ruleset;
  const newRuleset = startCaseKeys(rest);
  timeFields.forEach(field => {
    const value = newRuleset[field];
    newRuleset[field] = isInteger(value) ? formatDateTime(value) : value;
  });
  boolFields.forEach(field => {
    if (newRuleset[field]) {
      newRuleset[field] = capitalize(String(newRuleset[field]));
    }
  });
  return newRuleset;
};
