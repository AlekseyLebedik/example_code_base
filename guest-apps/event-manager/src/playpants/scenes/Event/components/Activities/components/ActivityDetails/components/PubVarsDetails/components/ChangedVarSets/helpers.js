import flatMap from 'lodash/flatMap';
import sortBy from 'lodash/sortBy';
import uniq from 'lodash/uniq';

export const getVarSetRowData = variableSets =>
  variableSets &&
  flatMap(
    variableSets.map(varSet =>
      sortBy(
        uniq([
          ...Object.keys(varSet.liveVariables || {}),
          ...Object.keys(varSet.variables),
        ])
      )
        .map(variable => ({
          variable,
          liveValue: varSet.liveVariables && varSet.liveVariables[variable],
          oldValue: varSet.oldVariables ? varSet.oldVariables[variable] : '---',
          newValue: varSet.variables[variable],
          context: varSet.context,
          groupId: varSet.group_id,
          namespace: varSet.namespace,
        }))
        .filter(variable => variable.newValue !== undefined)
    )
  );
