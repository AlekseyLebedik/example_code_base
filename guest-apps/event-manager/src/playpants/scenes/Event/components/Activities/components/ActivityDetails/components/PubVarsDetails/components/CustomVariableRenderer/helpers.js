export const getVariableMapping = activitySettings =>
  activitySettings.find(setting => setting.type === 'pubvars').variable_mapping;

export const printVariableMapping = (variable, variableMapping) =>
  variableMapping && variableMapping[variable]
    ? `${variableMapping[variable]} (${variable})`
    : variable;
