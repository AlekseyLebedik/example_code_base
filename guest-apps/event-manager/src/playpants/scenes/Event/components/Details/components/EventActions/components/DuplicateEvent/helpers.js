import { NEW_ACTIVITY_SETTINGS } from 'playpants/constants/activities';

export const formatAchievementEngine = activity => {
  const { ae } = NEW_ACTIVITY_SETTINGS;
  const {
    ruleset_to_activate: { label },
  } = activity;
  return {
    ...ae,
    ruleset_to_duplicate: label,
  };
};

export const formatActivity = (activity, type) => {
  switch (type) {
    case 'ae':
      return formatAchievementEngine(activity);
    default:
      return activity;
  }
};

export function redirectToDuplicatedEvent(baseUrl, { id }, history) {
  history.replace(`${baseUrl}events/${id}`);
}
