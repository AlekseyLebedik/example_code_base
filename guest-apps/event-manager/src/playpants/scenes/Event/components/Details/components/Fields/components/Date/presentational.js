import React from 'react';
import PropTypes from 'prop-types';

import DateTimePicker from 'dw/core/components/DateTimePickerV2/LocalizedFormFieldDateTimePicker';
import styles from './index.module.css';

const DateStateless = ({
  onSave,
  date,
  minDate,
  maxDate,
  disabled,
  label,
  placeholder,
  clearable,
}) => (
  <DateTimePicker
    input={{
      value: date,
      label,
      disabled,
      minDate,
      maxDate,
      clearable,
      InputLabelProps: { shrink: true },
      InputProps: { disabled, placeholder },
      classes: { input: styles.dateInput, dialog: styles.dialog },
      onChange: onSave,
    }}
  />
);

DateStateless.propTypes = {
  onSave: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  minDate: PropTypes.oneOfType([PropTypes.number, PropTypes.object]).isRequired,
  maxDate: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  placeholder: PropTypes.string,
  date: PropTypes.number,
  clearable: PropTypes.bool,
};

DateStateless.defaultProps = {
  placeholder: null,
  date: null,
  maxDate: null,
  clearable: false,
};

export default DateStateless;
