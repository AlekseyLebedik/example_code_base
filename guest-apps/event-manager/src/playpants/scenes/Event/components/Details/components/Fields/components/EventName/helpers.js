import find from 'lodash/find';
import isEmpty from 'lodash/isEmpty';

export const maxLength = (input, max) =>
  input && input.length > max
    ? {
        error: true,
        helperText: `Must be ${max} characters or less`,
      }
    : {};

export const required = input =>
  input && input.length ? {} : { error: true, helperText: 'Required' };

// find first failing validation (order-sensitive)
export const validateEventName = validations =>
  validations && find(validations, val => !isEmpty(val));
