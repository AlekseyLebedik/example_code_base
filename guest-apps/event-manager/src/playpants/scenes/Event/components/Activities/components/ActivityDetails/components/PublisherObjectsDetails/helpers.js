import { getNowTimestamp } from 'playpants/helpers/dateTime';
import { prettyPrint } from 'playpants/helpers/json';

// formats properties object needed for column layout and upload data needed for File Storage
export const formatPubObjectFormData = (
  { fileData, fileName, acl, expiresOn, category, groupID, checksumType },
  owner
) => {
  const properties = {
    acl,
    category,
    owner,
    groupID: groupID || '0',
    expiresOn: expiresOn || '0',
    created: getNowTimestamp(),
    modified: getNowTimestamp(),
    name: fileName || fileData.file.name,
    contentLength: fileData.file.size,
    extraData: null,
    checksumType,
  };
  const uploadData = {
    file: fileData.file,
    file_name: properties.name,
    type: 'publisher_objects',
    owner_id: properties.owner,
    uploaded_at: properties.created,
    properties: prettyPrint(properties),
  };

  const formData = new FormData();
  Object.entries(uploadData).forEach(([key, val]) => formData.append(key, val));
  return formData;
};

// adds data fileId and properties to update the current activity
export const formatActivityPayload = (data, currentActivity) => ({
  ...currentActivity,
  activity: prettyPrint({
    files: [...currentActivity.activity.files, data.id],
    objects: [...currentActivity.activity.objects, data.properties],
  }),
});
