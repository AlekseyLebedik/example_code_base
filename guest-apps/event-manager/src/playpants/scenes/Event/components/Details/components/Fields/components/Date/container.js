import React from 'react';
import PropTypes from 'prop-types';
import { getNowTimestamp } from 'playpants/helpers/dateTime';

import DateStateless from './presentational';

const Date = props => {
  const { eventData: event, type, onSave: saveDate } = props;

  const onSave = e => saveDate(type, e === '' ? null : e.unix());

  const newProps = {
    ...props,
    clearable: props.clearable && !props.detachedEvent,
    date: event[type],
    maxDate: type === 'publish_at' ? event.end_at : null,
    minDate: type === 'publish_at' ? getNowTimestamp() : event.publish_at,
    onSave,
  };

  return <DateStateless {...newProps} />;
};

Date.propTypes = {
  clearable: PropTypes.bool,
  detachedEvent: PropTypes.bool.isRequired,
  eventData: PropTypes.object.isRequired,
  onSave: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
};
Date.defaultProps = {
  clearable: false,
};

export default Date;
