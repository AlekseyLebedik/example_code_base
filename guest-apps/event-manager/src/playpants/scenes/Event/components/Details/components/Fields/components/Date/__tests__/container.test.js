import React from 'react';
import { shallow } from 'enzyme';

import { eventDatesProps as props } from 'playpants/testUtils/eventProps';
import * as DT from 'dw/core/helpers/date-time';
import Date from '../container';

describe('Date', () => {
  DT.getNowTimestamp = jest.fn(() => 1548547200);
  const root = shallow(<Date {...props} label="From" />);

  it('loads the Date component correctly', () => {
    root.setProps({
      eventData: { ...props.event, publish_at: 1542825180 },
      type: 'publish_at',
    });
    expect(root).toMatchSnapshot();
  });

  it('change to end at', () => {
    root.setProps({
      eventData: {
        ...props.event,
        publish_at: 1542825180,
        end_at: 1543825180,
      },
      type: 'end_at',
    });
    expect(root).toMatchSnapshot();
  });
});
