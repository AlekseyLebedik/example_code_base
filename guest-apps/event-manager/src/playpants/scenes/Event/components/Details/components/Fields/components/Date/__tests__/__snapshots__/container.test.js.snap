// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date change to end at 1`] = `
<DateStateless
  allowDetachedEvents={false}
  badgeCount={
    Object {
      "discussion": 2,
      "tasks": 0,
    }
  }
  baseUrl="/event-manager/3/"
  changeMainTab={[MockFunction]}
  classes={Object {}}
  clearable={true}
  conflicts={Array []}
  currentUser={
    Object {
      "id": 1,
      "name": "Initial User",
    }
  }
  date={1543825180}
  dateTime={[MockFunction]}
  deleteEvent={[MockFunction]}
  detachedEvent={false}
  disabled={false}
  editAuths={[MockFunction]}
  editEvent={[MockFunction]}
  event={
    Object {
      "data": Object {
        "activities": Array [
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "language": "sp",
                  "text": "Hola Amigo :D",
                },
              ],
            },
            "exec_order": 0,
            "id": 13,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "motd",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "key": "weapon_type",
                  "value": "Gun",
                },
                Object {
                  "key": "attachment_icons",
                  "value": Array [],
                },
                Object {
                  "key": "invalid",
                  "value": "test",
                },
              ],
              "name": "Double XP",
              "template_id": "double_xp",
              "version": "v1.0.4",
            },
            "exec_order": 1,
            "id": 12,
            "publish_on": "on_start",
            "title_envs": Array [
              4,
            ],
            "type": "pyscript",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "ruleset_to_activate": Object {
                "activationTimestamp": null,
                "code": "idiwoq03292kj4lsl",
                "codeHash": "4mhdBfYBdYmeco1ve62h0b6rue8=",
                "codeSignatureTimestamp": 1565841879,
                "code_signature": "v/kWwu8MBe1GGSxK64a3OqKynmKpGJYGlylUMT0asJ+ZHiuonlDAl2igzISRqgMuPWJf/HqCxvAbxuAMxq+YCg==",
                "creationTimestamp": 1565841879,
                "isActive": true,
                "label": "stronghold-5",
                "lastUpdateTimestamp": 1565842668,
              },
            },
            "exec_order": 0,
            "id": 1,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "ae",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "deploy": Array [
                Object {
                  "build_uid": "e36576c1fb46e4d60c9ccf1cb5ae2f01dfcd96c6e226b424c715aec84155bd9b",
                  "target_name": "uidev:dev(mp)",
                  "uid": "tp_deployment_1",
                },
              ],
              "modify": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa3",
                  "uid": "modify_1",
                  "user_params": Object {},
                },
              ],
              "undeploy": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa2",
                  "uid": "undeploy_1",
                },
              ],
            },
            "exec_order": 0,
            "id": 2,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "tp_deployment",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
        ],
        "authorizations": Array [],
        "authorizers": Array [],
        "auto_tags": Array [
          "PS4",
        ],
        "created_at": 1560983839,
        "created_by": Object {
          "id": 1,
          "name": "Initial User",
        },
        "end_at": 1561243001,
        "env_type": "Development",
        "event_type": "event-manager",
        "id": 2,
        "is_schedule": false,
        "locked_by": 1,
        "manual_tags": Array [
          "PS4",
        ],
        "note": "testing notes",
        "platforms": Array [
          "PS",
        ],
        "project": 1,
        "publish_at": 1561057200,
        "status": "open",
        "task": null,
        "title": "Test Event",
        "updated_at": 1560985103,
        "updated_by": Object {
          "id": 1,
          "name": "Initial User",
        },
      },
      "loading": false,
    }
  }
  eventData={
    Object {
      "data": Object {
        "activities": Array [
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "language": "sp",
                  "text": "Hola Amigo :D",
                },
              ],
            },
            "exec_order": 0,
            "id": 13,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "motd",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "key": "weapon_type",
                  "value": "Gun",
                },
                Object {
                  "key": "attachment_icons",
                  "value": Array [],
                },
                Object {
                  "key": "invalid",
                  "value": "test",
                },
              ],
              "name": "Double XP",
              "template_id": "double_xp",
              "version": "v1.0.4",
            },
            "exec_order": 1,
            "id": 12,
            "publish_on": "on_start",
            "title_envs": Array [
              4,
            ],
            "type": "pyscript",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "ruleset_to_activate": Object {
                "activationTimestamp": null,
                "code": "idiwoq03292kj4lsl",
                "codeHash": "4mhdBfYBdYmeco1ve62h0b6rue8=",
                "codeSignatureTimestamp": 1565841879,
                "code_signature": "v/kWwu8MBe1GGSxK64a3OqKynmKpGJYGlylUMT0asJ+ZHiuonlDAl2igzISRqgMuPWJf/HqCxvAbxuAMxq+YCg==",
                "creationTimestamp": 1565841879,
                "isActive": true,
                "label": "stronghold-5",
                "lastUpdateTimestamp": 1565842668,
              },
            },
            "exec_order": 0,
            "id": 1,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "ae",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "deploy": Array [
                Object {
                  "build_uid": "e36576c1fb46e4d60c9ccf1cb5ae2f01dfcd96c6e226b424c715aec84155bd9b",
                  "target_name": "uidev:dev(mp)",
                  "uid": "tp_deployment_1",
                },
              ],
              "modify": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa3",
                  "uid": "modify_1",
                  "user_params": Object {},
                },
              ],
              "undeploy": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa2",
                  "uid": "undeploy_1",
                },
              ],
            },
            "exec_order": 0,
            "id": 2,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "tp_deployment",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
        ],
        "authorizations": Array [],
        "authorizers": Array [],
        "auto_tags": Array [
          "PS4",
        ],
        "created_at": 1560983839,
        "created_by": Object {
          "id": 1,
          "name": "Initial User",
        },
        "end_at": 1561243001,
        "env_type": "Development",
        "event_type": "event-manager",
        "id": 2,
        "is_schedule": false,
        "locked_by": 1,
        "manual_tags": Array [
          "PS4",
        ],
        "note": "testing notes",
        "platforms": Array [
          "PS",
        ],
        "project": 1,
        "publish_at": 1561057200,
        "status": "open",
        "task": null,
        "title": "Test Event",
        "updated_at": 1560985103,
        "updated_by": Object {
          "id": 1,
          "name": "Initial User",
        },
      },
      "end_at": 1543825180,
      "loading": false,
      "publish_at": 1542825180,
    }
  }
  eventId={2}
  eventUrl="/event-manager/3/events/2"
  handleSetPrimaryTab={[MockFunction]}
  handleSetSecondaryTab={[MockFunction]}
  hasEndDate={false}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  isConfigured={
    Object {
      "stories": true,
    }
  }
  isEventManagerEvent={true}
  isLoadingEventsPage={false}
  label="From"
  loadDiscussion={[MockFunction]}
  loadEvent={[MockFunction]}
  maxDate={null}
  minDate={1542825180}
  onCloseRequestErrorDialog={[MockFunction]}
  onDeleteEvent={[MockFunction]}
  onSave={[Function]}
  permissions={
    Object {
      "achievementWritePermission": true,
      "adminPermission": true,
      "eventWritePermission": true,
      "publisherObjectsWritePermission": true,
      "pubstorageWritePermission": true,
      "pubvarsWritePermission": true,
      "pyscriptWritePermission": true,
      "staffUser": true,
      "tpDeploymentWritePermission": true,
      "wipPermission": true,
    }
  }
  placeholder={null}
  primaryTab="details"
  removeActivitySelection={[MockFunction]}
  requestErrorDialogOpen={
    Object {
      "open": false,
    }
  }
  saveEdit={[MockFunction]}
  secondaryTab="tasks"
  selectedTab="activities"
  setEndDate={[MockFunction]}
  status={
    Object {
      "allowAuths": true,
      "canCancel": true,
      "canDelete": true,
      "canOpen": false,
      "failed": false,
      "locked": false,
      "name": "Open",
      "readOnly": false,
      "showAuths": false,
      "showTemplates": true,
    }
  }
  storiesData={Array []}
  tabUrl="/event-manager/3/events/2/activities"
  type="end_at"
  updateUserProfileSetting={[MockFunction]}
  user={
    Object {
      "actions": Object {
        "createUserProfileSetting": [MockFunction],
        "updateUserProfileSetting": [MockFunction],
      },
    }
  }
/>
`;

exports[`Date loads the Date component correctly 1`] = `
<DateStateless
  allowDetachedEvents={false}
  badgeCount={
    Object {
      "discussion": 2,
      "tasks": 0,
    }
  }
  baseUrl="/event-manager/3/"
  changeMainTab={[MockFunction]}
  classes={Object {}}
  clearable={true}
  conflicts={Array []}
  currentUser={
    Object {
      "id": 1,
      "name": "Initial User",
    }
  }
  date={1542825180}
  dateTime={[MockFunction]}
  deleteEvent={[MockFunction]}
  detachedEvent={false}
  disabled={false}
  editAuths={[MockFunction]}
  editEvent={[MockFunction]}
  event={
    Object {
      "data": Object {
        "activities": Array [
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "language": "sp",
                  "text": "Hola Amigo :D",
                },
              ],
            },
            "exec_order": 0,
            "id": 13,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "motd",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "key": "weapon_type",
                  "value": "Gun",
                },
                Object {
                  "key": "attachment_icons",
                  "value": Array [],
                },
                Object {
                  "key": "invalid",
                  "value": "test",
                },
              ],
              "name": "Double XP",
              "template_id": "double_xp",
              "version": "v1.0.4",
            },
            "exec_order": 1,
            "id": 12,
            "publish_on": "on_start",
            "title_envs": Array [
              4,
            ],
            "type": "pyscript",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "ruleset_to_activate": Object {
                "activationTimestamp": null,
                "code": "idiwoq03292kj4lsl",
                "codeHash": "4mhdBfYBdYmeco1ve62h0b6rue8=",
                "codeSignatureTimestamp": 1565841879,
                "code_signature": "v/kWwu8MBe1GGSxK64a3OqKynmKpGJYGlylUMT0asJ+ZHiuonlDAl2igzISRqgMuPWJf/HqCxvAbxuAMxq+YCg==",
                "creationTimestamp": 1565841879,
                "isActive": true,
                "label": "stronghold-5",
                "lastUpdateTimestamp": 1565842668,
              },
            },
            "exec_order": 0,
            "id": 1,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "ae",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "deploy": Array [
                Object {
                  "build_uid": "e36576c1fb46e4d60c9ccf1cb5ae2f01dfcd96c6e226b424c715aec84155bd9b",
                  "target_name": "uidev:dev(mp)",
                  "uid": "tp_deployment_1",
                },
              ],
              "modify": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa3",
                  "uid": "modify_1",
                  "user_params": Object {},
                },
              ],
              "undeploy": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa2",
                  "uid": "undeploy_1",
                },
              ],
            },
            "exec_order": 0,
            "id": 2,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "tp_deployment",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
        ],
        "authorizations": Array [],
        "authorizers": Array [],
        "auto_tags": Array [
          "PS4",
        ],
        "created_at": 1560983839,
        "created_by": Object {
          "id": 1,
          "name": "Initial User",
        },
        "end_at": 1561243001,
        "env_type": "Development",
        "event_type": "event-manager",
        "id": 2,
        "is_schedule": false,
        "locked_by": 1,
        "manual_tags": Array [
          "PS4",
        ],
        "note": "testing notes",
        "platforms": Array [
          "PS",
        ],
        "project": 1,
        "publish_at": 1561057200,
        "status": "open",
        "task": null,
        "title": "Test Event",
        "updated_at": 1560985103,
        "updated_by": Object {
          "id": 1,
          "name": "Initial User",
        },
      },
      "loading": false,
    }
  }
  eventData={
    Object {
      "data": Object {
        "activities": Array [
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "language": "sp",
                  "text": "Hola Amigo :D",
                },
              ],
            },
            "exec_order": 0,
            "id": 13,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "motd",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "inputs": Array [
                Object {
                  "key": "weapon_type",
                  "value": "Gun",
                },
                Object {
                  "key": "attachment_icons",
                  "value": Array [],
                },
                Object {
                  "key": "invalid",
                  "value": "test",
                },
              ],
              "name": "Double XP",
              "template_id": "double_xp",
              "version": "v1.0.4",
            },
            "exec_order": 1,
            "id": 12,
            "publish_on": "on_start",
            "title_envs": Array [
              4,
            ],
            "type": "pyscript",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "ruleset_to_activate": Object {
                "activationTimestamp": null,
                "code": "idiwoq03292kj4lsl",
                "codeHash": "4mhdBfYBdYmeco1ve62h0b6rue8=",
                "codeSignatureTimestamp": 1565841879,
                "code_signature": "v/kWwu8MBe1GGSxK64a3OqKynmKpGJYGlylUMT0asJ+ZHiuonlDAl2igzISRqgMuPWJf/HqCxvAbxuAMxq+YCg==",
                "creationTimestamp": 1565841879,
                "isActive": true,
                "label": "stronghold-5",
                "lastUpdateTimestamp": 1565842668,
              },
            },
            "exec_order": 0,
            "id": 1,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "ae",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
          Object {
            "activity": Object {
              "deploy": Array [
                Object {
                  "build_uid": "e36576c1fb46e4d60c9ccf1cb5ae2f01dfcd96c6e226b424c715aec84155bd9b",
                  "target_name": "uidev:dev(mp)",
                  "uid": "tp_deployment_1",
                },
              ],
              "modify": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa3",
                  "uid": "modify_1",
                  "user_params": Object {},
                },
              ],
              "undeploy": Array [
                Object {
                  "build_uid": "84fe6b23-9eef-4fcd-b350-cef71c228eaa2",
                  "uid": "undeploy_1",
                },
              ],
            },
            "exec_order": 0,
            "id": 2,
            "publish_on": "on_start",
            "title_envs": Array [
              1,
            ],
            "type": "tp_deployment",
            "updated_by": Object {
              "id": 1,
              "name": "Initial User",
            },
          },
        ],
        "authorizations": Array [],
        "authorizers": Array [],
        "auto_tags": Array [
          "PS4",
        ],
        "created_at": 1560983839,
        "created_by": Object {
          "id": 1,
          "name": "Initial User",
        },
        "end_at": 1561243001,
        "env_type": "Development",
        "event_type": "event-manager",
        "id": 2,
        "is_schedule": false,
        "locked_by": 1,
        "manual_tags": Array [
          "PS4",
        ],
        "note": "testing notes",
        "platforms": Array [
          "PS",
        ],
        "project": 1,
        "publish_at": 1561057200,
        "status": "open",
        "task": null,
        "title": "Test Event",
        "updated_at": 1560985103,
        "updated_by": Object {
          "id": 1,
          "name": "Initial User",
        },
      },
      "loading": false,
      "publish_at": 1542825180,
    }
  }
  eventId={2}
  eventUrl="/event-manager/3/events/2"
  handleSetPrimaryTab={[MockFunction]}
  handleSetSecondaryTab={[MockFunction]}
  hasEndDate={false}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  isConfigured={
    Object {
      "stories": true,
    }
  }
  isEventManagerEvent={true}
  isLoadingEventsPage={false}
  label="From"
  loadDiscussion={[MockFunction]}
  loadEvent={[MockFunction]}
  maxDate={null}
  minDate={1548547200}
  onCloseRequestErrorDialog={[MockFunction]}
  onDeleteEvent={[MockFunction]}
  onSave={[Function]}
  permissions={
    Object {
      "achievementWritePermission": true,
      "adminPermission": true,
      "eventWritePermission": true,
      "publisherObjectsWritePermission": true,
      "pubstorageWritePermission": true,
      "pubvarsWritePermission": true,
      "pyscriptWritePermission": true,
      "staffUser": true,
      "tpDeploymentWritePermission": true,
      "wipPermission": true,
    }
  }
  placeholder={null}
  primaryTab="details"
  removeActivitySelection={[MockFunction]}
  requestErrorDialogOpen={
    Object {
      "open": false,
    }
  }
  saveEdit={[MockFunction]}
  secondaryTab="tasks"
  selectedTab="activities"
  setEndDate={[MockFunction]}
  status={
    Object {
      "allowAuths": true,
      "canCancel": true,
      "canDelete": true,
      "canOpen": false,
      "failed": false,
      "locked": false,
      "name": "Open",
      "readOnly": false,
      "showAuths": false,
      "showTemplates": true,
    }
  }
  storiesData={Array []}
  tabUrl="/event-manager/3/events/2/activities"
  type="publish_at"
  updateUserProfileSetting={[MockFunction]}
  user={
    Object {
      "actions": Object {
        "createUserProfileSetting": [MockFunction],
        "updateUserProfileSetting": [MockFunction],
      },
    }
  }
/>
`;
